addons/point_of_sale/models/pos_config.py

```diff
@@ -614,7 +610,6 @@ def _compute_local_data_integrity(self):
 
     def write(self, vals):
         if 'iface_tipproduct' in vals and not vals['iface_tipproduct']:
-            vals['tip_product_id'] = False
```

Comment:
You should also keep this,

When reactivating the functionality it should look after tip product and add it back

Reply:
Yes we expect to clear when desactivated and add it back when activated